core/src/main/scala/kafka/coordinator/transaction/TransactionLog.scala

```diff
@@ -0,0 +1,269 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.coordinator.transaction
+
+import kafka.common.{KafkaException, MessageFormatter}
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.protocol.types.Type._
+import org.apache.kafka.common.protocol.types.{ArrayOf, Field, Schema, Struct}
+import java.io.PrintStream
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+
+import org.apache.kafka.common.record.CompressionType
+
+import scala.collection.mutable
+
+/*
+ * Messages stored for the transaction topic represent the pid and transactional status of the corresponding
+ * transactional id, which have versions for both the key and value fields. Key and value
+ * versions are used to evolve the message formats:
+ *
+ * key version 0:               [transactionalId]
```

Comment:
Hmm, in the design doc, the key for transactional status message is the following. Is the doc outdated? Is it true that we are combining the transactional status message and the TransactionalId mapping message into a single one?

Key => Version Type PID
  Version => 0 (int16)
  Type => 1 (int16)
  PID => int64